lib/protobuf/decoder.ex

```diff
@@ -234,7 +234,15 @@ defmodule Protobuf.Decoder do
       prop
 
     embedded_msg = decode(bin, type)
-    val = if map?, do: %{embedded_msg.key => embedded_msg.value}, else: embedded_msg
+
+    val =
+      if map?,
+        do: %{
+          (embedded_msg.key || map_default(prop, :key)) =>
+            embedded_msg.value || map_default(prop, :value)
+        },
+        else: embedded_msg
```

Comment:
Nit but we'll probably want a `do`/`end` here rather than the inline `do:`.

Also, is this gonna be fine with booleans?